client/src/pages/Admin/UserDetails.jest.spec.js

```diff
@@ -0,0 +1,61 @@
+import { mountQuasar } from "@quasar/quasar-app-extension-testing-unit-jest";
+import UserDetails from "./UserDetails.vue";
+
+import * as All from "quasar";
+
+const components = Object.keys(All).reduce((object, key) => {
+  const val = All[key];
+  if (val.component?.name != null) {
+  object[key] = val;
+  }
+  return object;
+}, {});
+
+const query = jest.fn();
+describe('User Details page mount', () => {
+  const wrapperFactory = (userId) => {
+    return  mountQuasar(UserDetails, {
+      quasar: {
+        components
+      },
+      mount: {
+        type: 'full',
+        mocks: {
+          $t: token => token,
+          $apollo: {
+            query
+          },
+          $route: {
+            params: {
+              id: userId
+            }
+          }
+        },
+        stubs: ["router-link"]
+      }
+    });
+  }
+
+  it ('mounts without errors', () => {
+    expect(wrapperFactory(0)).toBeTruthy();
+  });
+
+  it('queries for a specific user', () => {
+    query.mockClear();
+    const wrapper = wrapperFactory(1);
+    expect(wrapper).toBeTruthy();
+
+    expect(
+      wrapper
+        .vm
+        .$options
+        .apollo
+        .user
+        .variables
+        .bind(wrapper.vm)().id
+    ).toBe(1)
+  });
+
+  //! to test how the view handles the data returned, use wrapper.setData(user: {...})
```

Comment:
Should populate the user roles field and test that it renders a list / placeholder if the list is empty. 

(And any other conditional logic in the template, but I think this is the big one)

Reply:
I'm struggling to get data mocked for this for now, but I'll keep trying. Unless [the documentation](https://vue-test-utils.vuejs.org/guides/getting-started.html) I'm reading is the wrong documentation, I'm finding it to be a bit frustrating because it recommends using methods that are deprecated, like `find()`.

Reply:
It turns out the way I was mocking data was fine, but the assertion wasn't working. I ended up utilizing `wrapper.text()` to check for conditional text being present under different circumstances. The unit test has been extended to check for the following conditional logic in the template:

* A user with no roles
* An application administrator
* A user with no name